lib/modules/measure/MeasureService.ts

```diff
@@ -237,6 +224,48 @@ export class MeasureService {
               })
           );
         }
+
+        for (const [measuredAt, assetState] of Object.entries(
+          assetMeasuresStates
+        )) {
+          const measureNames = [];
+
+          for (const measure of Object.values(assetState._source.measures)) {
+            if (measure.measuredAt === Number(measuredAt)) {
+              measureNames.push(measure.name);
+            }
+          }
+
+          const event: AssetHistoryEventMeasure = {
+            measure: {
+              names: measureNames,
+            },
+            name: "measure",
+          };
+
+          const changes = objectDiff(
+            originalAssetMetadata,
+            asset._source.metadata
+          );
+
+          if (changes.length !== 0) {
+            (event as unknown as AssetHistoryEventMetadata).metadata = {
+              names: changes,
+            };
+          }
```

Comment:
I'm confused @Aschen @fmauNeko 
This code means, if a metadata was edited and that there's multiple assetStates, each asset-history document event will have a metadata property. Where, maybe, only one (the first / the last) should have.
It's no big deal, but I find it weird. What do you think ?

Reply:
Very good remark.

Actually if a business rule edit the asset metadata after some computation on a mesure then the metadata property should be set on the according asset-history document (the one containing the measure as well)

This would need to think about an framework API to bound a measure with a metadata edition.

I think for now you can just save the metadata with the last document, edit documentation accordingly to explain this behavior and add a comment for the futur

Reply:
There's currently no test for this behaviour. I presume I should write one ?

Reply:
Right your presume young padawan

Reply:
A test, I shall write